### ISSUE TITLE
[BUG] Have 'nan' values when using 'energy distance' loss

### ISSUE THREAD

**Issue by thelittlewen** [jeanfeydy/geomloss#88]
**- bug description**

When _use_keops==True_, the program of the program pass the _if use_keops_ condition **(notice that this part don't have _torch.clamp_min_ protection)**
```
# geomloss.utils.py
def distances(x, y, use_keops=False):
    if use_keops:
        return squared_distances(x, y, use_keops=use_keops).sqrt()
    else:
        return torch.sqrt(torch.clamp_min(squared_distances(x, y), 1e-8))
```
and goes to: 
```
# geomloss.utils.py
def squared_distances(x, y, use_keops=False):

    if use_keops and keops_available:
        if x.dim() == 2:
            x_i = LazyTensor(x[:, None, :])  # (N,1,D)
            y_j = LazyTensor(y[None, :, :])  # (1,M,D)
        elif x.dim() == 3:  # Batch computation
            x_i = LazyTensor(x[:, :, None, :])  # (B,N,1,D)
            y_j = LazyTensor(y[:, None, :, :])  # (B,1,M,D)
        else:
            print("x.shape : ", x.shape)
            raise ValueError("Incorrect number of dimensions")

        return ((x_i - y_j) ** 2).sum(-1)

    else:
        if x.dim() == 2:
            D_xx = (x * x).sum(-1).unsqueeze(1)  # (N,1)
            D_xy = torch.matmul(x, y.permute(1, 0))  # (N,D) @ (D,M) = (N,M)
            D_yy = (y * y).sum(-1).unsqueeze(0)  # (1,M)
        elif x.dim() == 3:  # Batch computation
            D_xx = (x * x).sum(-1).unsqueeze(2)  # (B,N,1)
            D_xy = torch.matmul(x, y.permute(0, 2, 1))  # (B,N,D) @ (B,D,M) = (B,N,M)
            D_yy = (y * y).sum(-1).unsqueeze(1)  # (B,1,M)
        else:
            print("x.shape : ", x.shape)
            raise ValueError("Incorrect number of dimensions")

        return D_xx - 2 * D_xy + D_yy
```

but when keops_available is not available, it return using D_xx - 2 * D_xy + D_yy, which could return negative number. And the sqrt() operation on negative number leads to nan value.

**- suggested solution:**

make sure user' keops is available when they set use_keops=True (maybe it is set autonomously), or set a protection here:
```
def distances(x, y, use_keops=False):
    if use_keops:
        #return squared_distances(x, y, use_keops=use_keops).sqrt()
        sq_dist = squared_distances(x, y, use_keops=use_keops)
        sq_dist = torch.clamp_min(sq_dist, 0)
        return sq_dist.sqrt()
    else:
        return torch.sqrt(torch.clamp_min(squared_distances(x, y), 1e-8))

```
